client/widgets/CArtifactHolder.cpp

```diff
@@ -945,3 +943,102 @@ bool CArtifactsOfHero::SCommonPart::Artpos::valid()
 	assert(AOH && art);
 	return art == AOH->curHero->getArt(slotID);
 }
+
+CArtPlace::CArtPlace(Point position, const CArtifactInstance * Art) : ourArt(Art)
+{
+	pos += position;
+	pos.w = pos.h = 44;
+}
+
+void CArtPlace::clickLeft(tribool down, bool previousState)
+{
+	LRClickableAreaWTextComp::clickLeft(down, previousState);
+}
+
+void CArtPlace::clickRight(tribool down, bool previousState)
+{
+	LRClickableAreaWTextComp::clickRight(down, previousState);
+}
+
+CCommanderArtPlace::CCommanderArtPlace(Point position, const CGHeroInstance * commanderOwner, ArtifactPosition artSlot, const CArtifactInstance * Art) : CArtPlace(position, Art), commanderOwner(commanderOwner), commanderSlotID(artSlot.num)
```

Comment:
Yeah, this class constructor takes may arguments, but I guess it is ok, considering it does everything without relying on CArtifactsOfHero magic